internal/ignition/ignition.go

```diff
@@ -963,23 +970,55 @@ func isBaremetalProvisioningConfig(file *config_latest_types.File) bool {
 	return strings.Contains(file.Node.Path, "baremetal-provisioning-config")
 }
 
-func fileToBMH(file *config_latest_types.File) (*bmh_v1alpha1.BareMetalHost, error) {
+func removeProvisioningPausedAnnotation(file *config_latest_types.File) error {
+	provisioning := &metal3iov1alpha1.Provisioning{}
+	if err := deserializeToObject(file, provisioning); err != nil {
```

Comment:
This will cause us problems in future, because the API definition will have to be kept up to date every time a field is added to the Provisioning resource, and we will certainly forget that this is here and then wonder why the new data is getting silently dropped.

This might still be the best thing to do in the short term, but it's not a great substitute for a long-term fix that doesn't require adding this annotation at all.

Reply:
How long do you think it will be until we get a proper long-term fix?
Do you think doing this in the controller against the actual cluster with a patch API call will be any more maintainable?

Reply:
We got a successful test on https://github.com/openshift/installer/pull/8391 but there's a few moving parts so it might take a minute to land.

Reply:
Using a patch API call would avoid the out-of-date-API-definition problem.